### DISCUSSION TITLE
Query Listener

### DISCUSSION THREAD

**Discussion by duranguenze** [wheels-dev/wheels#1452] [Q&A]
How i implement the query listener, in lucee i use to apply in the application.cfc, but cfwheels don't take of that. It helpme to debug the old issues, and register the querys called.

**Comment by chapmandu** [comment]
Depending which event method you're using in Application.cfc, use the equivalent file in the /events/ directory.

**Reply by duranguenze** [reply]
In application i use to catch every query calls with this.
<cfset this.query.listener = {
            before = function (caller,args) {
                return args;
            }
            ,after = function () {
                arguments.result.recordCount eq 0
                arguments.args.sql
           }
}>

How i can catch that events.

**Reply by neokoenig** [reply]
If you're using `this` you should be able to add it to `/config/app.cfm`

**Reply by duranguenze** [reply]
In the aplication.cfc, i put it in the onrequest, for log every query executed, i try it in the app.cfm, but just log the login in the start.
